zipline/utils/events.py

```diff
@@ -487,24 +474,16 @@ class NthTradingDayOfWeek(TradingDayOfWeekRule):
     def get_first_trading_day_of_week(dt, env):
         prev = dt
         dt = env.previous_trading_day(dt)
-        # If we're on the first trading day of the TradingEnvironment,
-        # calling previous_trading_day on it will return None, which
-        # will blow up when we try and call .date() on it. The first
-        # trading day of the env is also the first trading day of the
-        # week(in the TradingEnvironment, at least), so just return
-        # that date.
-        if dt is None:
-            return prev
-        while dt.date().weekday() < prev.date().weekday():
+        # Traverse backward until we hit a week border, then jump back to the
+        # previous trading day.
+        while dt and dt.date().weekday() < prev.date().weekday():
```

Comment:
Not sure how inner this loop is, but I think we can get `weekday()` from `dt` itself.